naucse_render/course.py

```diff
@@ -0,0 +1,193 @@
+    elif isinstance(value, (list, tuple)):
+        return [encode_dates(v) for v in value]
+    elif isinstance(value, (str, int, bool, type(None))):
+        return value
+    raise TypeError(value)
+
+
+# XXX: Clarify the version
+
+def get_course(course_slug: str, *, path='.', version):
+    """Get information about the course as a JSON-compatible dict"""
+
+    base_path = Path(path).resolve()
+
+    if version <= 0:
+        raise ValueError(f'Version {version} is not supported')
+
+    if course_slug == 'lessons':
+        info = get_canonical_lessons_info(path)
+        path_parts = None
+    else:
+        parts = course_slug.split('/')
+        if len(parts) == 1 or (len(parts) == 2 and parts[0] == 'courses'):
+            path_parts = 'courses', parts[-1]
+        elif len(parts) == 2:
+            path_parts = 'runs', *parts
+        else:
+            raise ValueError(f'Invalid course slug')
+
+        info = read_yaml(
+            base_path, *path_parts, 'info.yml',
+            source_key='source_file',
+        )
+
+    info.pop('meta', None)
+    info.pop('canonical', None)
+
+    base_slug = info.get('derives', None)
+    if base_slug:
+        base_course = read_yaml(base_path, 'courses', base_slug, 'info.yml')
+    else:
+        base_course = {}
+
+    # Rename "plan" to "sessions"
+    for d in info, base_course:
+        if 'plan' in d:
+            d['sessions'] = d.pop('plan')
+
+    if 'long_description' in info:
+        info['long_description'] = convert_markdown(info['long_description'])
+
+    # Rename the text field "time" to "time_description"
+    if 'time' in info:
+        info['time_description'] = info.pop('time')
+
+    for session in info['sessions']:
+        session['source_file'] = info['source_file']
+        base = session.pop('base', None)
+        if base:
+            for base_session in base_course['sessions']:
+                if base_session['slug'] == base:
+                    break
+            else:
+                raise ValueError(f'Session {session} not found in base course')
+            session.update(merge_dict(base_session, session))
+        for material in session.get('materials', []):
+            update_material(material, vars=info.get('vars'), path=base_path)
+
+        if 'description' in session:
+            session['description'] = convert_markdown(session['description'])
+
+        if path_parts:
+            # Get coverpage content
+            page_path = base_path.joinpath(
+                *path_parts, 'sessions', session['slug']
+            )
+            if page_path.exists():
+                session['pages'] = {}
+                for page_md_path in page_path.glob('*.md'):
+                    session['pages'][page_md_path.stem] = {
+                        'content': convert_markdown(page_md_path.read_text()),
+                    }
+
+    result = encode_dates(info)
+    return {
+        'api_version': [0, 0],
```

Comment:
what does this actually mean?

Reply:
0.0, just in a format that's easier to parse.
Do you think that's a bad idea?

Reply:
the comment helped.